File: App/src/main/java/it/gruppoinfor/home2work/database/RoutePointRepo.java
package it.gruppoinfor.home2work.database;


import android.arch.persistence.room.Room;
import android.content.Context;

import com.arasthel.asyncjob.AsyncJob;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;

import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;
import it.gruppoinfor.home2workapi.HomeToWorkClient;
import it.gruppoinfor.home2workapi.model.User;

public class RoutePointRepo {

    private DBApp dbApp;

    public RoutePointRepo(Context context) {
        dbApp = Room.databaseBuilder(context, DBApp.class, "home2work")
                .fallbackToDestructiveMigration()
                .build();
    }

    public void insert(RoutePointEntity routePointEntity) {
        AsyncJob.doInBackground(() -> dbApp.routePointDAO().insert(routePointEntity));
    }

    public void getAllUserLocations(OnSuccessListener<List<RoutePointEntity>> onSuccessListener, OnFailureListener onFailureListener) {
        dbApp.routePointDAO().getAllUserPoints(HomeToWorkClient.getUser().getId())
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(onSuccessListener::onSuccess, throwable -> onFailureListener.onFailure(new Exception(throwable)));
    }

    public void deleteAllUserLocations(OnSuccessListener<Integer> onSuccessListener) {
        AsyncJob.doInBackground(() -> {
            int deletedRows = dbApp.routePointDAO().deleteAll(HomeToWorkClient.getUser().getId());
            AsyncJob.doOnMainThread(() -> onSuccessListener.onSuccess(deletedRows));

        });
    }

    public void deleteAllUserLocations(long userId) {
        AsyncJob.doInBackground(() -> dbApp.routePointDAO().deleteAll(userId));
    }

    public void insert(RoutePointEntity routePointEntity, OnSuccessListener<Long> onSuccessListener) {
        AsyncJob.doInBackground(() -> {
            Long id = dbApp.routePointDAO().insert(routePointEntity);
            AsyncJob.doOnMainThread(() -> onSuccessListener.onSuccess(id));

        });
    }


}

File: App/src/main/java/it/gruppoinfor/home2work/database/RoutePointRepo.kt
package it.gruppoinfor.home2work.database


import android.content.Context
import com.arasthel.asyncjob.AsyncJob
import com.google.android.gms.tasks.OnFailureListener
import com.google.android.gms.tasks.OnSuccessListener
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import it.gruppoinfor.home2workapi.HomeToWorkClient

class RoutePointRepo constructor(ctx: Context) {

    private val locationsDatabase: LocationsDatabase = LocationsDatabase.getInstance(ctx)!!

    fun insert(routePointEntity: RoutePointEntity) {
        AsyncJob.doInBackground { locationsDatabase.routePointDAO().insert(routePointEntity) }
    }

    fun getAllUserLocations(onSuccessListener: OnSuccessListener<List<RoutePointEntity>>, onFailureListener: OnFailureListener) {

        locationsDatabase.routePointDAO().getAllUserPoints(HomeToWorkClient.getUser().id!!)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe({ onSuccessListener.onSuccess(it) }, { onFailureListener.onFailure(Exception(it)) })
    }

    fun deleteAllUserLocations(onSuccessListener: OnSuccessListener<Int>) {
        AsyncJob.doInBackground {
            val deletedRows = locationsDatabase.routePointDAO().deleteAll(HomeToWorkClient.getUser().id!!)
            AsyncJob.doOnMainThread { onSuccessListener.onSuccess(deletedRows) }

        }
    }

    fun deleteAllUserLocations(userId: Long) {
        AsyncJob.doInBackground { locationsDatabase.routePointDAO().deleteAll(userId) }
    }

    fun insert(routePointEntity: RoutePointEntity, onSuccessListener: OnSuccessListener<Long>) {
        AsyncJob.doInBackground {
            val id = locationsDatabase.routePointDAO().insert(routePointEntity)
            AsyncJob.doOnMainThread { onSuccessListener.onSuccess(id) }
        }
    }


}
